scripts/generate-docs.js

```diff
@@ -266,27 +292,45 @@ if (require.main === module) {
 	require("./_command")({
 		exec: generate,
 		async watch() {
-			let timeout, genPromise
-			function updateGenerator() {
-				if (timeout == null) return
-				clearTimeout(timeout)
-				genPromise = new Promise((resolve) => {
-					timeout = setTimeout(function() {
-						timeout = null
-						resolve(makeGenerator().then((g) => g.generate()))
-					}, 100)
-				})
+			const generator = (await makeGenerator())
+
+			function isLayoutFile(relative) {
+				return relative === "layout.html"
+			}
+
+			function isNavigationFile(relative) {
+				return ["nav-guides.md", "nav-methods.md"].includes(relative)
+			}
+
+			async function updateGenerator() {
+				generator.generate()
 			}
 
 			async function updateFile(file) {
-				if ((/^layout\.html$|^archive$|^nav-/).test(file)) {
-					updateGenerator()
+				const relative = path.relative(r("docs"), file)
+				if (isLayoutFile(relative)) {
+					generator.setLayout(await loadLayoutFile())
+					generator.generate()
+				} else if (isNavigationFile(relative)) {
+					const [guides, methods] = await loadNavigationFiles()
+					generator.setGuides(guides)
+					generator.setMethods(methods)
+					generator.generate()
+				} else {
+					generator.generateSingle(file)
 				}
-				(await genPromise).generateSingle(file)
 			}
 
 			async function removeFile(file) {
-				(await genPromise).eachTarget(file, (dest) => fs.unlink(dest))
+				const relative = path.relative(r("docs"), file)
+				if (isLayoutFile(relative)) {
+					throw `Error: the layout file "${relative}" is required!`
+				} else if (isNavigationFile(relative)) {
+					throw `Error: the navigation file "${relative}" is required!`
+				}
+				const relativeDist = relative.replace(/\.md$/, ".html")
+				generator.eachTarget(relativeDist, (dest) => fs.unlink(dest))
```

Comment:
Same as before, different method.

```suggestion
				await generator.eachTarget(relativeDist, (dest) => fs.unlink(dest))
```